boto/endpoints.py

```diff
@@ -0,0 +1,200 @@
+        loads from the new endpoint format. It should only be used for
+        partition metadata and partition specific service metadata.
+
+        :type partition_name: str
+        :param partition_name: The name of the partition to search for.
+
+        :returns: Partition info from the new endpoints format.
+        :rtype: dict or None
+        """
+        for partition in self._endpoint_data['partitions']:
+            if partition['partition'] == partition_name:
+                return partition
+        raise ValueError(
+            "Could not find partition data for: %s" % partition_name)
+
+    def _endpoint_prefix(self, service_name):
+        """Given a boto2 service name, get the endpoint prefix."""
+        return self._endpoint_prefix_map.get(service_name, service_name)
+
+    def _service_name(self, endpoint_prefix):
+        """Given an endpoint prefix, get the boto2 service name."""
+        return self._service_name_map.get(endpoint_prefix, endpoint_prefix)
+
+
+class StaticEndpointBuilder(object):
+    """Builds a static mapping of endpoints in the legacy format."""
+
+    def __init__(self, resolver):
+        """
+        :type resolver: BotoEndpointResolver
+        :param resolver: An endpoint resolver.
+        """
+        self._resolver = resolver
+
+    def build_static_endpoints(self, service_names=None):
+        """Build a set of static endpoints in the legacy boto2 format.
+
+        :param service_names: The names of the services to build. They must
+            use the names that boto2 uses, not boto3, e.g "ec2containerservice"
+            and not "ecs". If no service names are provided, all available
+            services will be built.
+
+        :return: A dict consisting of::
+            {"service": {"region": "full.host.name"}}
+        """
+        if service_names is None:
+            service_names = self._resolver.get_available_services()
+
+        static_endpoints = {}
+        for name in service_names:
+            endpoints_for_service = self._build_endpoints_for_service(name)
+            if endpoints_for_service:
+                # It's possible that when we try to build endpoints for
+                # services we get an empty hash.  In that case we don't
+                # bother adding it to the final list of static endpoints.
+                static_endpoints[name] = endpoints_for_service
+        self._handle_special_cases(static_endpoints)
+        return static_endpoints
+
+    def _build_endpoints_for_service(self, service_name):
+        # Given a service name, 'ec2', build a dict of
+        # 'region' -> 'hostname'
+        endpoints = {}
+        regions = self._resolver.get_all_available_regions(service_name)
+        for region_name in regions:
+            endpoints[region_name] = self._resolver.resolve_hostname(
+                service_name, region_name)
+        return endpoints
+
+    def _handle_special_cases(self, static_endpoints):
```

Comment:
Thinking about this again. I think it if fine keeping it here. The customization is specifically needed for that static endpoint file. The data for cloudsearchdomain does not exist in the endpoints file so it would be strange to inject it there.